consensus/src/quorum_store/utils.rs

```diff
@@ -378,14 +378,26 @@ impl ProofQueue {
         &mut self,
         excluded_batches: &HashSet<BatchInfo>,
         max_txns: u64,
+        max_unique_txns: u64,
         max_bytes: u64,
         return_non_full: bool,
     ) -> (Vec<ProofOfStore>, bool) {
         let mut ret = vec![];
         let mut cur_bytes = 0;
-        let mut cur_txns = 0;
+        let mut cur_unique_txns = 0;
+        let mut cur_all_txns = 0;
         let mut excluded_txns = 0;
         let mut full = false;
+        // Set of all the excluded transactions and all the transactions included in the result
+        let mut included_and_excluded_txns = HashSet::new();
```

Comment:
nit: would `filter_txns` be easier to understand?

Reply:
Done.

Reply:
Done.